src/ripple/basics/Slice.h

```diff
@@ -140,6 +164,28 @@ class Slice
     {
         return data_ + size_;
     }
+
+    /** Return a "sub slice" of given length starting at the given position
+
+        Note that the subslice encompasses the range [pos, pos + rcount),
+        where rcount is the smaller of count and size() - pos.
+
+        @param pos position of the first character
+        @count requested length
+
+        @returns The requested subslice, if the request is valid.
+        @throws std::out_of_range if pos > size()
+     */
+    Slice
+    substr(
+        std::size_t pos,
+        std::size_t count = std::numeric_limits<std::size_t>::max()) const
+    {
```

Comment:
Consider making this `constexpr` and `pos` default to zero, as `string_view::substr` does.

Reply:
Please update `Slice_test.cpp` to cover new functions.